Assets/Scripts/Models/Buildable/Components/FluidConnection.cs

```diff
@@ -0,0 +1,305 @@
+
+            if (IsConsumer && Requires != null)
+            {
+                yield return LocalizationTable.GetLocalization("fluid_input_status", powerColor, Requires.Rate);
+            }
+
+            if (IsProducer && Requires != null)
+            {
+                yield return LocalizationTable.GetLocalization("fluid_output_status", powerColor, Requires.Rate);
+            }
+
+            if (IsAccumulator && Provides != null)
+            {
+                yield return LocalizationTable.GetLocalization("fluid_accumulated_fraction", AccumulatedAmount, Provides.Capacity);
+            }
+        }
+
+        public void Reconnect()
+        {
+            if (Reconnecting != null)
+            {
+                Reconnecting();
+            }
+        }
+
+        protected override void Initialize()
+        {
+            componentRequirements = Requirements.Fluid;
+
+            if (ParamsDefinitions == null)
+            {
+                // don't need definition for all furniture, just use defaults
+                ParamsDefinitions = new FluidConnectionParameterDefinitions();
+            }
+
+            // need to keep accumulator current charge in parameters
+
+            // TODO: need to prevent defining same type more than once?
+            if (Provides != null)
+            {
+                if (Provides.Capacity > 0)
+                {
+                    CurrentAccumulatorCharge.SetValue(0f);
+                    CurrentAccumulatorChargeIndex.SetValue(0);
+                }
+            }
+
+            IsRunning = false;
+
+            OnReconnecting();
+
+            ParentFurniture.Removed += FluidConnectionRemoved;           
+        }
+
+        private void FluidConnectionRemoved(Furniture obj)
+        {
+            World.Current.FluidNetwork.Unplug(this);
+            ParentFurniture.Removed -= FluidConnectionRemoved;
+        }
+
+        private void OnReconnecting()
+        {
+            // TODO: Make this not a Universal Connection
+            World.Current.FluidNetwork.PlugIn(this);
+        }
+
+        [Serializable]
+        [JsonObject(MemberSerialization.OptOut)]
+        public class Info
+        {
+            [XmlAttribute("rate")]
+            public float Rate { get; set; }
+
+            [XmlAttribute("capacity")]
+            public float Capacity { get; set; }
+
+            [XmlAttribute("capacityThresholds")]
+            public int CapacityThresholds { get; set; }
+
+            [XmlElement("Param")]
+            public List<ParameterCondition> ParamConditions { get; set; }            
+        }
+
+        [Serializable]
+        [JsonObject(MemberSerialization.OptOut)]
+        public class FluidConnectionParameterDefinitions
+        {
+            // constants for parameters
+            public const string CurAccumulatorChargeParamName = "pow_accumulator_charge";
+            public const string CurAccumulatorIndexParamName = "pow_accumulator_index";
```

Comment:
I realize this, as I mentioned in the original post, I haven't yet changed all the assorted names left over from its origins as PowerConnection, there's probably quite a few more beyond those two, and intend to change them all once I've hammered out the functionality, and gotten further input on said functionality.